### ISSUE TITLE
TensorBoard fails grouping summaries by names

### ISSUE THREAD

**Issue by dalgu90** [tensorflow/tensorflow#8591]
Hi. Using Tensorboard, I found that it fails grouping scalar summaries by its name.

As I know, summaries are grouped by their name scopes, which are delimited by slash('/').
However, for some cases, the group to be bound together would be separated.

Here's a reproducible code that causes the same error, and the resulting tensorboard page.

```
import tensorflow as tf
import os

if not os.path.exists('./temp'):
    os.mkdir('./temp')
summary_writer = tf.summary.FileWriter('./temp', None)

summary = tf.Summary()
summary.value.add(tag='train/accuracy', simple_value=1.0)
summary.value.add(tag='train/cross_entropy', simple_value=1.0)
summary.value.add(tag='train/learning_rate', simple_value=1.0)
summary.value.add(tag='train_image/foo', simple_value=1.0)

for i in range(10):
    summary_writer.add_summary(summary, i)
summary_writer.flush()
```

![image](https://cloud.githubusercontent.com/assets/13655756/24163367/b4673d48-0ead-11e7-8478-b0733581b90e.png)

### Environment info
Operating System: Ubuntu 14.04
Installed version of CUDA and cuDNN: No cuda related, I guess.
TensorFlow version: Only confirmed in the nightly build, Linux-GPU version. But I think it will occur in other versions as well.

Thank you for your hard work. :^)

**Comment by wookayin** [comment]
It seems that tag names are sorted ignoring non-alphabet characters (such as `_` and `/`) and they are grouped according to the sort order :-)

**Comment by skye** [comment]
@dandelionmane I'm assigning to you for now, please reassign if necessary.

**Comment by chihuahua** [comment]
This is fixed by 1968b8b3f9316a8327176e25211c8cc2b01459bc.
